onebot/plugins/psa.py

```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8 -*-
+"""
+================================================
+:mod:`onebot.plugins.psa` PSA
+================================================
+
+This plugin allows admins to send broadcasts
+"""
+
+import irc3
+from irc3 import plugin
+from irc3.plugins.command import command
+
+
+@plugin
+class PSAPlugin(object):
+    """PSA Plugin"""
+
+    requires = [
+        'irc3.plugins.command',
+    ]
+
+    def __init__(self, bot):
+        self.bot = bot
+        self.channels = set()
+
+    @irc3.event(irc3.rfc.JOIN)
+    def register_channel(self, mask, channel, **kwargs):
```

Comment:
You're right, of course. As it turns out, there's the `irc3.plugins.userlist` plugin that can manage the up-to-date channel list (I was initially puzzled why `self.bot.channels` was not populated, but it turns out that it requires this plugin).